beta-src/src/state/game/game-api-slice.ts

```diff
@@ -293,13 +305,116 @@ const drawBuilds = (state) => {
   }
 };
 
+const updateUnitsDisbanding = (state) => {
+  const {
+    data: {
+      data: { contextVars, currentOrders, units },
+    },
+    territoriesMeta,
+    ordersMeta,
+  }: {
+    data;
+    territoriesMeta;
+    ordersMeta;
+  } = current(state);
+  if (currentOrders && contextVars) {
+    const { context } = contextVars;
+    if (context.phase === "Retreats") {
+      const tMeta: TerritoriesMeta = territoriesMeta;
+      const terrMetaEntries = Object.entries(tMeta);
+      const overtakenTerritories = terrMetaEntries.filter(([id, val]) => {
+        return (
+          val.countryID === context.countryID &&
+          val.countryID !== val.ownerCountryID
+        );
+      });
+
+      let userDisbandingUnits;
+      let territoryKeys;
+      console.log("ot", overtakenTerritories);
+
+      if (overtakenTerritories && overtakenTerritories.length > 0) {
+        const terrKeyTempArray: string[] = [];
+        const disbandingUnitsTempArray: CurrentOrder[] = [];
+        overtakenTerritories.forEach(([terrKey, val]) => {
+          const { unitID } = val;
+          console.log("OM", ordersMeta);
+          const overtakenUnitsInOrder = currentOrders.find(
+            (o) =>
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                o.type === "Disband") ||
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                ordersMeta[o.id] &&
+                ordersMeta[o.id].update.type === "Disband"),
+          );
```

Comment:
Rather than looping through each territories to see if there is an associated unit, why not just loop through the ordersMeta which already has an associated unit? This way you don't have to loop through unnecessary territories and use `.find()` inside of each territory.

Reply:
This gives me easy access to the territoryKey which is needed for making updates like destroying or building a unit post take over

Reply:
This gives me access to territoryKeys which allow me to set a unit upon ready.

Reply:
What are the territoryKeys? You should be able to get detailed territory data by using the ordersMeta key (which is the orderID). What if the user has 30 territories and the current respective order for the unit you're trying to find is the 29th? Then you need to loop through 29 territories and use .find on each of them before finding your unit. 

What if there are 30 territories and the user has 12 units and the current order is the last unit in the 29th territory? Then you're processing 319 times without really needing to.

Also, I'm surprised typescript didn't catch lines 344, 347, and 350, as those properties may be undefined. 

You can replace `ordersMeta[o.id] && ordersMeta[o.id].update.type === "Disband"` with `ordersMeta[id]?.update?type === "Disband"` and combine with object destructuring to make this code less verbose.

Reply:
One other thing I noticed is that "Disband" is capitalized, but the `OrderType` type definitions are all lowercase. Have you added the `disband` type to the `OrderType` type? If not you should and keep it consistent (lowercase).